### PR TITLE
Common components from the explore-dash

### PR DESCRIPTION
**Pull request by Syn-McJ**
`feature-explore-dash` has some components that can be used in other branches for other features. Those are better to be merged to the master now.

## Issue being fixed or feature implemented
- Added Hilt to the Wallet module. This is required for DI to work in any of the feature modules.
- Provided `Configuration` for injection of shared preferences.
- `FragmentViewBindingDelegate` for simplifying view bindings in fragments. Usage:
`private val binding by viewBinding(FragmentSearchBinding::bind)`
- `SegmentedPicker` - implements [Segmented Picker](https://www.figma.com/file/azdJACb5WmivxYVhB5q46F/Android-%7C-Design-system?node-id=669%3A335) from the design system.
- `OffsetDialogFragment` for a bottom sheet dialog that has 20dp offset from the top (for example, [Filters Dialog](https://www.figma.com/file/aghEG2PAqmXghOWi5BoHg5/Android-%7C-DashPay-Master?node-id=107%3A11959)). Has rounded background and wraps content if the amount of content in the dialog isn't enough for a full screen.
- `RadioGroupAdapter` simplifies creating a list of [RadioButtions](https://www.figma.com/file/azdJACb5WmivxYVhB5q46F/Android-%7C-Design-system?node-id=683%3A694) from the design system.
- `OptionPickerDialog` generic dialog with search and list of radio buttons. For example, used by [territory picker](https://www.figma.com/file/aghEG2PAqmXghOWi5BoHg5/Android-%7C-DashPay-Master?node-id=107%3A11018) (although in the design it looks like a fragment, it's implemented as a bottom sheet dialog).
- `NavigationExtensions.kt` provides `safeNavigate` extension method for fragments which helps to avoid "Navigation action/destination xyz cannot be found from the current destination" errors.
- `CheckBox` style implements [Checkbox](https://www.figma.com/file/azdJACb5WmivxYVhB5q46F/Android-%7C-Design-system?node-id=679%3A545) from the design system. The old CheckBox renamed to `OldCheckBox` to avoid confusion.
- Various text styles to implement [typography](https://www.figma.com/file/azdJACb5WmivxYVhB5q46F/Android-%7C-Design-system?node-id=479%3A147) from the design system.
- Various button styles to implement [Buttons](https://www.figma.com/file/azdJACb5WmivxYVhB5q46F/Android-%7C-Design-system?node-id=234%3A988) from the design system. Only a relatively small subset is implemented and we still have a lot of old styles for buttons, so it's better if any news styles are based on these.

## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [x] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: common/src/main/java/org/dash/wallet/common/services/analytics/AnalyticsConstants.kt [hunk 1]
```diff
@@ -1,17 +1,20 @@
 /*
- * Copyright 2021 Dash Core Group
  *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
+ *  * Copyright 2021 Dash Core Group.
```
Comment (HashEngineering) [inline common/src/main/java/org/dash/wallet/common/services/analytics/AnalyticsConstants.kt:3]: There are a few license headers that were indented.

`*  *`

File: wallet/src/de/schildbach/wallet/di/ConfigurationModule.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 Dash Core Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/di/ConfigurationModule.kt:6]: I wonder how many files have an Apache 2.0 License.

**Review by HashEngineering — APPROVED** [review]
Many licenses have `*  *`.

File: common/src/main/java/org/dash/wallet/common/services/analytics/AnalyticsConstants.kt (hunk 1 above)
Comment (Syn-McJ) [reply common/src/main/java/org/dash/wallet/common/services/analytics/AnalyticsConstants.kt:3]: I just copy-pasted this one from some other file. Will fix.

File: wallet/src/de/schildbach/wallet/di/ConfigurationModule.kt (hunk 2 above)
Comment (Syn-McJ) [reply wallet/src/de/schildbach/wallet/di/ConfigurationModule.kt:6]: This is the old template I used, will fix. Most of the DashPay files have Apache licenses.

**Review by hadia — APPROVED** [review]
Looks good 👌

**Pull request merged**